### ISSUE TITLE
Declaration with two DeviceGroups only deploys first group in the declaration

### ISSUE THREAD

**Issue by HidingUnderMyRock** [F5Networks/f5-declarative-onboarding#149]
### Environment
 * Declarative Onboarding Version: 1.13.0
 * BIG-IP Version: 15.1.0.2 and 14.1.2.3
 * BIG-IQ Version: 7.1.0

### Summary
A declaration that includes two DeviceGroup objects, one sync-failover and another sync-only, will only result in the first DeviceGroup in the declaration being deployed on the device.

### Steps To Reproduce
1. Submit the following declaration via BIG-IQ:
```json
{
    "class": "DO",
    "declaration": {
        "schemaVersion": "1.5.0",
        "class": "Device",
        "async": true,
        "Common": {
            "class": "Tenant",
            "hostname": "{{hostname}}.{{domain}}",
            "myLicense": {
                "class": "License",
                "licenseType": "licensePool",
                "unitOfMeasure": "yearly",
                "licensePool": "{{ltmLicensePool}}",
                "skuKeyword1": "{{ltmLicenseSku}}",
                "overwrite": true,
                "bigIpUsername": "{{user}}",
                "bigIpPassword": "{{password}}"
            },
            "myProvision": {
                "class": "Provision",
                "ltm": "nominal",
                "asm": "nominal",
                "avr": "nominal"
            },
            "myDbVariables": {
                "class": "DbVariables",
                "ui.advisory.enabled": "true",
                "ui.advisory.color": "red",
                "ui.advisory.text": "{{hostname}}.{{domain}}"
            },
            "myDns": {
                "class": "DNS",
                "nameServers": [
                    "10.0.0.2"
                ],
                "search": [
                    "{{domain}}"
                ]
            },
            "myNtp": {
                "class": "NTP",
                "servers": [
                    "pool.ntp.org"
                ],
                "timezone": "America/Chicago"
            },
            "ext_vlan": {
                "class": "VLAN",
                "interfaces": [
                    {
                        "name": "1.1",
                        "tagged": false
                    }
                ]
            },
            "int_vlan": {
                "class": "VLAN",
                "interfaces": [
                    {
                        "name": "1.2",
                        "tagged": false
                    }
                ]
            },
            "ext_self": {
                "class": "SelfIp",
                "address": "{{extIp}}/24",
                "vlan": "ext_vlan",
                "allowService": "none"
            },
            "int_self": {
                "class": "SelfIp",
                "address": "{{intIp}}/24",
                "vlan": "int_vlan"
            },
            "defaultRoute": {
                "class": "Route",
                "gw": "10.0.11.1",
                "network": "default"
            },
            "myConfigSync": {
                "class": "ConfigSync",
                "configsyncIp": "{{intIp}}"
            },
            "myFailoverUnicast": {
                "class": "FailoverUnicast",
                "address": "{{intIp}}"
            },
            "myDeviceTrust": {
                "class": "DeviceTrust",
                "localUsername": "{{user}}",
                "localPassword": "{{password}}",
                "remoteHost": "{{peerIp}}",
                "remoteUsername": "{{user}}",
                "remotePassword": "{{password}}"
            },
            "syncOnlyGroup": {
                "class": "DeviceGroup",
                "type": "sync-only",
                "owner": "{{hostname}}.{{domain}}",
                "members": [
                    "{{hostname}}.{{domain}}",
                    "{{hostname2}}.{{domain}}"
                ],
                "autoSync": true,
                "asmSync": true,
                "networkFailover": false
            },
            "syncFailoverGroup": {
                "class": "DeviceGroup",
                "type": "sync-failover",
                "owner": "{{hostname}}.{{domain}}",
                "members": [
                    "{{hostname}}.{{domain}}",
                    "{{hostname2}}.{{domain}}"
                ],
                "autoSync": true,
                "asmSync": false,
                "networkFailover": true
            },
            "admin": {
                "class": "User",
                "userType": "regular",
                "password": "{{password}}",
                "shell": "bash",
                "keys": [
                    "ssh-rsa **snip**"
                ]
            }
        }
    },
    "targetUsername": "{{user}}",
    "targetHost": "{{mgmtIp}}",
    "targetPassphrase": "{{tempPassword}}",
    "bigIqSettings": {
        "failImportOnConflict": false,
        "conflictPolicy": "USE_BIGIQ",
        "deviceConflictPolicy": "USE_BIGIP",
        "versionedConflictPolicy": "KEEP_VERSION",
        "clusterName": "{{clusterName}}",
        "useBigiqSync": true,
        "statsConfig": {
            "enabled": true,
            "zone": "default"
        }
    }
}
```

### Expected Behavior
The device should have both device groups deployed, one sync-failover group and another sync-only group.

### Actual Behavior
The declaration above will result in devices which only have the group `syncOnlyGroup.` If the order of the DeviceGroups sections are reversed in the declaration, placing the `syncFailoverGroup` first, the device will only have the `syncFailoverGroup.`

**Comment by dstokesf5** [comment]
Does the same declaration sent directly to the BIG-IP, instead of going through BIG-IQ, also fail?

**Comment by HidingUnderMyRock** [comment]
Yes, I just tested a declaration that contained two DeviceGroups directly to a BIG-IP and the result was only the first group was created.  This does not appear to be limited to declarations sent via BIG-IQ.

**Comment by dstokesf5** [comment]
Thank you for reporting this issue. I have reproduced it and added it to our internal product backlog as AUTOTOOL-1799.

**Comment by cbascom** [comment]
Any update on when we might get a fix for this? We can't use DO to set up HA when ASM is in use because of this issue. We need to be able to create an asm sync-only device group in addition to the sync-failover group in order to set up ASM synchronization correctly.

**Comment by dstokesf5** [comment]
A developer is currently looking into this issue, and we hope to have it resolved as part of the 1.18.0 release.

**Comment by dstokesf5** [comment]
This issue has been resolved as part of the 1.18.0 release.
